Open IPC Sooner

jenr24G opened the pull request:
Changes the server IPC listener to open before the server initialization, so the client can communicate with the server during initialization. Also adds a test to verify this functionality. Related: #225, #221

trevorbernard commented:
I haven't started the review yet but now that we open the domain socket right away, that also means it's going to accept updates from the file watcher. Is this a problem?

jenr24G commented:
> I haven't started the review yet but now that we open the domain socket right away, that also means it's going to accept updates from the file watcher. Is this a problem?

No, the file watcher is spawned by the `run()` method. I extracted the IPC code from the `select!` in the `run()` method and moved it to it's own task which I spawn just before calling `initialize()`
